cypress-io/cypress
# chore: adding command log to protocol

### Additional details


* Calls `protocol.commandLogAdded` when the `Cypress.on(log:added)` event is triggered.
* Calls `protocol.commandLogChanged` when the `Cypress.on(log:changed)` event is triggered.
* Updated tests that were spying/stubbing `Cypress.backend` to not log since that will now cause an infinite loop where we log the spy/stub and then call the backend again for the protocol which causes a log for the spy/stub and so on.

### Steps to test


- Clone the `cypress-services` repo and checkout `mschile/protocol/add_command_log`
  - Run `yarn`
  - Run `yarn watch-protocol`
- Set `CYPRESS_LOCAL_PROTOCOL_PATH` to the path to `cypress-services/packages/app-capture-protocol/dist/index.js`
- Clone the `cypress` repo  and checkout `mschile/protocol/add_command_log`
  - Run `yarn`
  - Execute `cypress run` on a project in record mode with `DEBUG` set to `cypress:*protocol*` and ensure the `command:log` events are added to the database.

### How has the user experience changed?


### PR Tasks


- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
